### DISCUSSION TITLE
How to add a constant block in Chatflow?

### DISCUSSION THREAD

**Discussion by weilun-enso** [langgenius/dify#9254] [Help]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I want to use a constant block that can be referenced in multiple other nodes. 
I have tried the following ways, but none worked conveniently:
- I tried setting it in input, but this is not so convenient as it gets lost every time I rerun the preview
- Conversation variables does not support paragraph like long variables as default
Is there any ways to achieve this currently? If not, I believe this can be a good feature to simplify our Chatflow graph.

Below is a mock example of how this can be useful:
# constant_node
"
[some ...
...
...
 paragraph]
"
# LLM1's system prompt
blablabla1
Context:
{constant_node}

# LLM2's system prompt
blablabla2
Context:
{constant_node}

# LLM3's system prompt
blablabla3
Context:
{constant_node}

This way, it's much easier to share the context part of all system prompts from different branches.

**Comment by Yevanchen** [comment]
The string of the session variable can store quite a large number of bytes. Could you further describe your needs? Rerunning the preview creates a new session; do you have any scenarios in mind for wanting a constant block that spans across sessions?

**Reply by weilun-enso** [reply]
Thanks for the reply! 

Like for example, I need a multiline context section that needs to be shared across multiple LLM nodes in the system prompt.
Say context section reads:
```
Your role is a writer, you need to follow the rules below:

- Never fabricate user's experience
- Make your language concise
- ...
```
Though theoretically, we could put these in a single line as a conv variable with '\n' as newline, but it won't be easily editable like the box for system prompts in an LLM node.


If I do this in the preview as an input, then each time I rerun preview by clicking the button, the input clears. It also clears it each time I reload the project. But this is some constant I need to use all the time. So it's better to keep it persistant.

Is everything a bit clearer now?

**Reply by weilun-enso** [reply]
@Yevanchen Any update on this? We're planning to use the service, but this is currently the blocking factor.
<img width="382" alt="Screenshot 2024-10-16 at 12 14 20 PM" src="https://github.com/user-attachments/assets/ee5a6c6f-06b1-41e2-b208-00749db6951d">
As shown in the screenshot, there can only be one line for default value of a conversation variable.
Even using the conversation variable but with a editing box to support multiline input like system prompt would be sufficient.

**Reply by weilun-enso** [reply]
Any update on this? We really need this feature for our project. I think the easiest way is simply to support multilines for default value of conversation variables. (Even the description field has it!) @Yevanchen
